### PR TITLE
Upgrade to Django 3.2

### PR DESCRIPTION
**Pull request by noliveleger**
## Description
Upgrade code base to use Django 3.2 LTS

## Additional info
- Set `DEFAULT_AUTO_FIELD`
- `django.contrib.postgres.fields.JSONField` deprecated. Replaced with `models.JSONField` [[ref]](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/fields/#jsonfield)
- `Constance` stuck at 2.8.0 in pipy (2.9.0 is 3.10 compatible) 
- `django-trench` is only marked as compatible with Python 3.9 and Django 3.0. Also stuck at 0.3.0 in pipy but GitHub project only has 0.3.1. Another PR has been created to updated it #3793
- Get rid of `prod.*` file for pip dependencies (Make it consistent with KoBoCAT) 

# Related issues
https://github.com/kobotoolbox/kobocat/pull/815

### CODE REVIEW

File: dependencies/pip/requirements.in [hunk 1]
```diff
@@ -13,7 +13,7 @@
 -e git+https://github.com/kobotoolbox/ssrf-protect@9b97d3f0fd8f737a38dd7a6b64efeffc03ab3cdd#egg=ssrf_protect
 
 # Regular PyPI packages
-Django>=2.2,<2.3
+Django==3.2
```
Comment (bufke) [inline dependencies/pip/requirements.in:16]: Should this be `>=3.2,<3.3`? This version exactly would have known security vulnerabilities in it.

**Comment by bufke** [comment]
Dockerfile has `COPY ./dependencies/pip/external_services.txt "/tmp/pip_dependencies.txt"` but that file no longer exists.

File: dependencies/pip/requirements.in (hunk 1 above)
Comment (noliveleger) [reply dependencies/pip/requirements.in:16]: I thought that `==3.2` would always have installed the latest patch version. I'm gonna change it to `>=3.2, <3.3` anyway.

**Comment by noliveleger** [comment]
> Dockerfile has `COPY ./dependencies/pip/external_services.txt "/tmp/pip_dependencies.txt"` but that file no longer exists.

:facepalm:, bad previous merge when we introduced multi builds.

**Review by bufke — APPROVED** [review]

**Pull request merged**